### PR TITLE
[Components with more than 1 service] LPD-13402 Refactor OAuth2AuthorizationExpandoPortalInstanceLifecycleListener component

### PR DESCRIPTION
**Pull request by MrJohn1911**
Previous pull: https://github.com/brianchandotcom/liferay-portal/pull/146360

Ticket: https://liferay.atlassian.net/browse/LPD-13402

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud-acceptance

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud-acceptance

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/java/com/liferay/osb/faro/web/internal/oauth2/provider/scope/spi/prefix/handler/OAuth2AuthorizationExpandoPrefixHandlerFactory.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.osb.faro.web.internal.oauth2.provider.scope.spi.prefix.handler;
```
Comment (dantewang) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/java/com/liferay/osb/faro/web/internal/oauth2/provider/scope/spi/prefix/handler/OAuth2AuthorizationExpandoPrefixHandlerFactory.java:6]: @MrJohn1911 It seems lots of oauth2 interfaces->implementations follow the package name pattern as in your last pull request, but that was rejected.

I wonder if we should just remove "spi" from the package name. The idea is, the interface is from module `oauth2-provider-scope-spi` whose bundle symbolic name is `com.liferay.oauth2.provider.scope.spi`. This symbolic name is also the common prefix of packages exported by that module. When we implement this interface, shouldn't the implementation's package be the impl modules' package prefix + the package part without the interface module's prefix?

That results in `com.liferay.osb.faro.web.internal.prefix.handler`. A similar recommendation is in source formatter document `package.markdown`.

Comment (MrJohn1911) [reply modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/java/com/liferay/osb/faro/web/internal/oauth2/provider/scope/spi/prefix/handler/OAuth2AuthorizationExpandoPrefixHandlerFactory.java:6]: Thanks @dantewang, now this is clear to me!

**Comment by MrJohn1911** [comment]
ci:test:sf

**Pull request closed without merging**